packages/rpc-core/src/rpc-methods/__tests__/get-token-accounts-by-delegate-test.ts

```diff
@@ -0,0 +1,387 @@
+                message: expect.any(String),
+                name: 'SolanaJsonRpcError',
+            });
+        });
+    });
+
+    describe('when called with a mint that has a delegated token account', () => {
+        it('returns RPC response with account info', async () => {
+            expect.assertions(1);
+            // Delegate for fixtures/spl-token-token-account-delegated.json
+            const delegate =
+                'G4QJANEpvEN8vLaaMZoWwZtqHfWxuWpd5RrVVYXPCgeL' as Base58EncodedAddress<'G4QJANEpvEN8vLaaMZoWwZtqHfWxuWpd5RrVVYXPCgeL'>;
+
+            // Mint at fixtures/spl-token-mint-account-with-delegated.json (mint for above delegated account)
+            const mint =
+                '4SspA9vWmizwcvngHTapwQtpnRrPf8V483giCSaCmy6M' as Base58EncodedAddress<'4SspA9vWmizwcvngHTapwQtpnRrPf8V483giCSaCmy6M'>;
+
+            const accountInfoPromise = rpc
+                .getTokenAccountsByDelegate(
+                    delegate,
+                    { mint },
+                    {
+                        encoding: 'base64',
+                    }
+                )
+                .send();
+
+            await expect(accountInfoPromise).resolves.toMatchObject({
+                value: [
+                    expect.objectContaining({
+                        account: expect.objectContaining({
+                            data: [expect.any(String), 'base64'],
+                            executable: expect.any(Boolean),
+                            lamports: expect.any(BigInt),
+                            owner: expect.any(String),
+                            rentEpoch: expect.any(BigInt),
+                            space: expect.any(BigInt),
+                        }),
+                    }),
+                ],
+            });
+        });
+    });
+
+    describe('when called with a mint that has no delegated token accounts', () => {
+        it('returns an empty list', async () => {
+            expect.assertions(1);
+            // Delegate for fixtures/spl-token-token-account-delegated.json
+            const delegate =
+                'G4QJANEpvEN8vLaaMZoWwZtqHfWxuWpd5RrVVYXPCgeL' as Base58EncodedAddress<'G4QJANEpvEN8vLaaMZoWwZtqHfWxuWpd5RrVVYXPCgeL'>;
+
+            // Mint at fixtures/spl-token-mint-no-token-accounts.json (not the same mint as above delegated account)
+            const mint =
+                'HWHfrWotTpaNArteqeYDziV1ZX9Lm7WV684NeUCwPPzj' as Base58EncodedAddress<'HWHfrWotTpaNArteqeYDziV1ZX9Lm7WV684NeUCwPPzj'>;
+
+            const accountInfoPromise = rpc.getTokenAccountsByDelegate(delegate, { mint }).send();
+
+            await expect(accountInfoPromise).resolves.toMatchObject({
+                value: [],
+            });
+        });
+    });
+
+    describe('when called with a program that is not a Token program', () => {
+        it('throws an error for unrecognized program', async () => {
```

Comment:
This one is kind of funny. So it will only recognize the two token programs for this call?